osgp/protocol-adapter-mqtt/osgp-protocol-simulator-mqtt/src/main/java/org/opensmartgridplatform/simulator/protocol/mqtt/Broker.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.simulator.protocol.mqtt;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import io.moquette.broker.Server;
+import io.moquette.broker.config.IConfig;
+import io.moquette.interception.AbstractInterceptHandler;
+import io.moquette.interception.messages.InterceptPublishMessage;
+import io.netty.buffer.ByteBuf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Moquette broker.
+ *
+ * Config is expected on the classpath at `config/moquette.conf`
+ */
+public final class Broker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Broker.class);
+    private static final String LISTENER_ID = "MoquetteBroker";
+
+    private static volatile boolean running;
+    private final IConfig config;
+
+    public Broker(final IConfig config) {
+        this.config = config;
+    }
+
+    public static boolean isRunning() {
+        return running;
+    }
+
+    public void start() {
+        if (isRunning()) {
+            LOG.warn("Broker already runnning, exiting");
+            return;
+        }
+        final Server server = new Server();
+        this.startServer(server);
+        handleShutdown(server);
+    }
+
+    private void startServer(final Server server) {
+        try {
+            server.startServer(this.config, Collections.singletonList(new AbstractInterceptHandler() {
+                @Override
+                public String getID() {
+                    return LISTENER_ID;
+                }
+
+                @Override
+                public void onPublish(final InterceptPublishMessage msg) {
+                    LOG.info(String.format("Broker received on topic: %s content: %s%n", msg.getTopicName(),
+                            new String(getBytes(msg), UTF_8)));
+                }
+            }));
+            running = true;
+            LOG.info("Broker started press [CTRL+C] to stop");
+        } catch (final IOException e) {
+            throw new RuntimeException("Could not start Moquette broker", e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00112)

Reply:
Propagated the IOException itself